File: leetcode/0521_longest_uncommon_subsequence.cpp
// Copyright (c) 2018 Jacopo Notarstefano

#include <algorithm>
#include <cassert>
#include <string>

using namespace std;

class Solution {
 public:
    int findLUSlength(string a, string b) {
        if (a == b) {
            return -1;
        } else {
            return max(a.length(), b.length());
        }
    }
};

int main() {
    auto solution = Solution();

    assert(3 == solution.findLUSlength("aba", "cdc"));
}

File: leetcode/0521_longest_uncommon_subsequence.py
# -*- coding: utf-8 -*-


class Solution:
    def findLUSlength(self, a, b):
        if a == b:
            return -1
        return max(len(a), len(b))


if __name__ == "__main__":
    solution = Solution()

    assert 3 == solution.findLUSlength("aba", "cdc")
